core/validatorapi/validatorapi.go

```diff
@@ -357,6 +359,69 @@ func (c Component) SubmitBeaconBlock(ctx context.Context, block *spec.VersionedS
 	return nil
 }
 
+// SubmitVoluntaryExit receives the partially signed voluntary exit.
+func (c Component) SubmitVoluntaryExit(ctx context.Context, ve *eth2p0.SignedVoluntaryExit) error {
+	vs, err := c.Validators(ctx, "head", []eth2p0.ValidatorIndex{ve.Message.ValidatorIndex})
+	if err != nil {
+		return err
+	}
+	validator, ok := vs[ve.Message.ValidatorIndex]
+	if !ok {
+		return errors.New("cannot find validator", z.U64("validator_index", uint64(ve.Message.ValidatorIndex)))
+	}
+
+	eth2Pubkey, err := validator.PubKey(ctx)
+	if err != nil {
+		return err
+	}
+
+	pubKey, err := core.PubKeyFromBytes(eth2Pubkey[:])
+	if err != nil {
+		return err
+	}
+
+	sigRoot, err := ve.Message.HashTreeRoot()
+	if err != nil {
+		return err
+	}
+
+	err = c.verifyParSig(ctx, core.DutyVoluntaryExit, ve.Message.Epoch, pubKey, sigRoot, ve.Signature)
+	if err != nil {
+		return err
+	}
+
+	data, err := json.Marshal(ve)
+	if err != nil {
+		return nil
+	}
+
+	parSigDate := core.ParSignedData{
+		Data:      data,
+		Signature: core.SigFromETH2(ve.Signature),
+		ShareIdx:  c.shareIdx,
+	}
+
+	parsigSet := core.ParSignedDataSet{
+		pubKey: parSigDate,
+	}
+
+	duty := core.Duty{
+		Type: core.DutyVoluntaryExit,
+		// By instructions of @corverroos I am putting an epoch into a field called Slot
+		// https://github.com/ObolNetwork/charon/pull/648#discussion_r886724647
```

Comment:
This comment doesn't seem very professional

Reply:
I am happy to remove it, and you add the code in a commit under your name.

If this is a bad decision, that you or the company should be ashamed of, please let me know why we are taking it, when there are better alternatives.

Reply:
so, do you want me to remove it?

Reply:
Done